pythran/pythonic/numpy/fft/c2c.hpp

```diff
@@ -0,0 +1,287 @@
+#include "pythonic/include/utils/array_helper.hpp"
+#include "pythonic/types/ndarray.hpp"
+#include "pythonic/builtins/None.hpp"
+#include "pythonic/numpy/concatenate.hpp"
+#include "pythonic/numpy/zeros.hpp"
+#include "pythonic/numpy/empty.hpp"
+
+
+#include <array>
+#include <cstring>
+#include <cmath>
+
+#include "pythonic/numpy/fft/pocketfft.hpp"
+
+PYTHONIC_NS_BEGIN
+
+namespace numpy
+{
+  namespace fft
+  {
+    using pocketfft::stride_t;
+    using pocketfft::shape_t;
+    using ldbl_t = typename std::conditional<sizeof(long double)==sizeof(double), double, long double>::type;
+
+    template <class T, class pS>
+    types::ndarray<typename std::enable_if<std::is_integral<T>::value, double>::type,
+                types::array<long, std::tuple_size<pS>::value>>
+    _copy_to_double(types::ndarray<T, pS> const &in_array)
+    {
+        auto out_shape = sutils::getshape(in_array);
+        size_t l = in_array.flat_size();
+        auto out_array = numpy::empty(out_shape, types::dtype_t<double>());
+        std::copy(in_array.buffer, in_array.buffer + l, out_array.buffer);
+        return out_array;
+    }
+
+    template <class T, class pS>
+    types::ndarray<typename std::enable_if<std::is_floating_point<T>::value, std::complex<T>>::type,
+                types::array<long, std::tuple_size<pS>::value>>
+    _copy_to_complex(types::ndarray<T, pS> const &in_array)
+    {
+        auto out_shape = sutils::getshape(in_array);
+        size_t l = in_array.flat_size();
+        auto out_array = numpy::empty(out_shape, types::dtype_t<typename std::complex<T>>());
+        std::copy(in_array.buffer, in_array.buffer + l, out_array.buffer);
+        return out_array;
+    }
+
+    template <class T, class pS>
+    types::ndarray<typename std::enable_if<std::is_integral<T>::value, std::complex<double>>::type,
+                types::array<long, std::tuple_size<pS>::value>>
+    _copy_to_complex(types::ndarray<T, pS> const &in_array)
+    {
+        auto out_shape = sutils::getshape(in_array);
+        size_t l = in_array.flat_size();
+        auto out_array = numpy::empty(out_shape, types::dtype_t<typename std::complex<double>>());
+        std::copy(in_array.buffer, in_array.buffer + l, out_array.buffer);
+        return out_array;
+    }
+
+    enum class Inorm: int
+    {
+        forward,
+        ortho,
+        backward,
+    };
+
+    Inorm _get_inorm(types::str const &norm, bool forward)
+    {
+        Inorm inorm;
+        if (norm == "ortho") {
+            inorm = Inorm::ortho;
+        } else if (!forward){
+            inorm = Inorm::backward;
+        } else {
+            inorm = Inorm::forward;
+        }
+        return inorm;
+    }
+
+    template<typename T> T norm_fct(Inorm inorm, size_t N)
+    {
+        switch (inorm)
+        {
+            case Inorm::ortho : return T(1/sqrt(ldbl_t(N)));
+            case Inorm::backward : return T(1/ldbl_t(N));
+            case Inorm::forward : return T(1);
```

Comment:
I did first use !types::is_complex<T> but then I need to add conditionals to distinguish between integer and floating scenarios, i.e. convert the integers to double, but not floats (because we want to preserve precision).